libethereum/ChainParams.cpp

```diff
@@ -107,6 +173,36 @@ ChainParams ChainParams::loadGenesisState(string const& _json, h256 const& _stat
 	return cp;
 }
 
+namespace
+{
+	string const c_parentHash = "parentHash";
+	string const c_coinbase = "coinbase";
+	string const c_author = "author";
+	string const c_difficulty = "difficulty";
+	string const c_gasLimit = "gasLimit";
+	string const c_gasUsed = "gasUsed";
+	string const c_timestamp = "timestamp";
+	string const c_extraData = "extraData";
+	string const c_mixHash = "mixHash";
+	string const c_nonce = "nonce";
+
+	set<string> const c_knownGenesisFields = {
+		c_parentHash, c_coinbase, c_author, c_difficulty, c_gasLimit, c_gasUsed, c_timestamp,
+		c_extraData, c_mixHash, c_nonce
+	};
+
+	void validateGenesis(js::mObject const& _genesis)
+	{
+		for (auto const& elm: _genesis)
+			if (c_knownGenesisFields.find(elm.first) == c_knownGenesisFields.end())
+			{
+				string const comment = "Unknown field in genesis config[genesis]: " + elm.first;
+				cerr << comment << "\n";
```

Comment:
Do we need this? Wouldn't exception handler output it, too, since we add it to exception?

Reply:
The exception handler currently does not output it. The handler is somewhere in boost_test.